### PR TITLE
add option to automatically update default user agent

### PR DESCRIPTION
**Pull request by henrik9999**
Adds an option which is enabled by default to update the default user agent on startup from this [list](https://github.com/jnrbsn/user-agents) idea was from AZ on discord.

I basically copied the extension update code, not really a Kotlin/Android dev, let me know if there is a better way to do something.

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/network/api/UserAgentGithubAPI.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Date
+import java.util.concurrent.TimeUnit
+
+internal class UserAgentGithubAPI {
+    private val networkService: NetworkHelper by injectLazy()
+    private val preferences: PreferencesHelper by injectLazy()
+
+    private var requiresFallbackSource = false
+
+    suspend fun getRandomUserAgent(): String? {
```
[the remark this answers is not in the document]
Comment (henrik9999) [reply app/src/main/java/eu/kanade/tachiyomi/network/api/UserAgentGithubAPI.kt:23]: That could maybe be an issue yeah, we could also just always take the first one i guess?

[the remark this answers is not in the document]
Comment (ghostbear) [reply app/src/main/java/eu/kanade/tachiyomi/network/api/UserAgentGithubAPI.kt:23]: I feel like the user could just do it manually at that point

[the remark this answers is not in the document]
Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/network/api/UserAgentGithubAPI.kt:23]: We could have a more detailed screen that has:

- Current set user agent (edit text)
- Reset option, if applicable
- List of user agents that can be copied or selected -- these can come from a list, if available

If we do want to fetch a list at runtime though, I'd want to at least fork that repo to the tachiyomiorg so we have better control over it.

[the remark this answers is not in the document]
Comment (Riztard) [reply app/src/main/java/eu/kanade/tachiyomi/network/api/UserAgentGithubAPI.kt:23]: i think manual random button would be better
just like manual input custom but easier

[the remark this answers is not in the document]
Comment (henrik9999) [reply app/src/main/java/eu/kanade/tachiyomi/network/api/UserAgentGithubAPI.kt:23]: Then I'll be closing this, I do not have an any intention for now to work on the other idea

**Pull request closed without merging**